packages/react-self-serializers/src/GraphQLComponent.tsx

```diff
@@ -1,30 +1,38 @@
 import React from 'react';
 
 import {useSerialized} from '@shopify/react-html';
-import {
-  ApolloProvider,
-  createGraphQLClient,
-  Options,
-} from '@shopify/react-graphql';
+import {ApolloProvider, createSsrExtractableLink} from '@shopify/react-graphql';
 import {useLazyRef} from '@shopify/react-hooks';
 
-interface Props extends Partial<Options> {
+interface Props {
   children?: React.ReactNode;
+  createClient(): import('apollo-client').ApolloClient<any>;
```

Comment:
❤️  for decoupling this and allow user to pass any client setup they would like. 
If only this was available an hour ago :p